File: STGPlugin/src/main/java/org/workcraft/plugins/stg/STGModule.java
package org.workcraft.plugins.stg;

import org.workcraft.Framework;
import org.workcraft.Module;
import org.workcraft.PluginManager;
import org.workcraft.dom.ModelDescriptor;
import org.workcraft.gui.propertyeditor.SettingsPage;
import org.workcraft.interop.Exporter;
import org.workcraft.interop.Importer;
import org.workcraft.plugins.interop.DotGExporter;
import org.workcraft.plugins.interop.DotGImporter;
import org.workcraft.plugins.stg.serialisation.ImplicitPlaceArcDeserialiser;
import org.workcraft.plugins.stg.serialisation.ImplicitPlaceArcSerialiser;
import org.workcraft.serialisation.xml.XMLDeserialiser;
import org.workcraft.serialisation.xml.XMLSerialiser;

public class STGModule implements Module {

	@Override
	public void init(Framework framework) {
		final PluginManager p = framework.getPluginManager();
		
		p.registerClass(ModelDescriptor.GLOBAL_SERVICE_HANDLE, new STGModelDescriptor());
		
		p.registerClass(XMLSerialiser.SERVICE_HANDLE, new ImplicitPlaceArcSerialiser());
		p.registerClass(XMLDeserialiser.SERVICE_HANDLE, new ImplicitPlaceArcDeserialiser());
		
		p.registerClass(Exporter.SERVICE_HANDLE, new DotGExporter());
		p.registerClass(Importer.SERVICE_HANDLE, new DotGImporter());
		
		p.registerClass(SettingsPage.SERVICE_HANDLE, new STGSettings());
	}

	@Override
	public String getDescription() {
		return "Signal Transition Graphs";
	}
}

File: STGPlugin21/src/main/scala/org/workcraft/plugins/stg21/StgModule.scala
package org.workcraft.plugins.stg21
import org.workcraft.Framework
import org.workcraft.dom.ModelDescriptor
import org.workcraft.plugins.interop.DotGImporter
import org.workcraft.interop.Importer
import org.workcraft.interop.Exporter
import org.workcraft.plugins.interop.DotGExporter
import org.workcraft.services.GlobalServiceProvider
import org.workcraft.services.Service
import org.workcraft.services.GlobalScope
import org.workcraft.services.NewModelService
import org.workcraft.services.NewModelImpl
import org.workcraft.services.Module

class StgModule extends Module {
  def name = "Signal Transition Graphs"
  def serviceProvider = StgServiceProvider
}

object StgServiceProvider extends GlobalServiceProvider {
  def implementations[T](service: Service[GlobalScope, T]) = service match {
    case NewModelService => List(NewStg)
    case _ => Nil
  }
}

object NewStg extends NewModelImpl {
  def name = "Signal Transition Graph"
  def create = StgModel.create
}

/*class StgModule extends Module {
	val getDescription : String = "Signal Transition Graphs"
	def init(framework : Framework) {
	  framework.getPluginManager().registerClass(ModelDescriptor.GLOBAL_SERVICE_HANDLE, StgModelDescriptor)
	  framework.getPluginManager().registerClass(Importer.SERVICE_HANDLE, DotGImporter)
	  framework.getPluginManager().registerClass(Exporter.SERVICE_HANDLE, DotGExporter)
	}
}
*/